csrc/preseg_passes/decompose_reshardings.cpp

```diff
@@ -245,7 +245,9 @@ void canonicalizeLoopDomain(TensorView* tv) {
 }
 
 void decomposeRowParallelLinearWithBias(Fusion* fusion) {
-  for (Expr* e : fusion->exprs()) {
+  // Iterate backwards over fusion expressions to avoid processing
+  // expressions that have been deleted.
```

Comment:
I'm sure you've tested this. Can you show me the failure pattern? IIUC, fusion->exprs() returns a **copy** of the original expressions so it sounds OK to traverse forward.

Reply:
Answered in a different comment on the test file, let's discuss there.